azurerm/resource_arm_route_test.go

```diff
@@ -35,6 +35,41 @@ func TestAccAzureRMRoute_basic(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMRoute_update(t *testing.T) {
+	resourceName := "azurerm_route.test"
+	ri := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testCheckAzureRMRouteDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMRoute_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMRouteExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "next_hop_type", "VnetLocal"),
+				),
+			},
+			{
+				Config: testAccAzureRMRoute_basicAppliance(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMRouteExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "next_hop_type", "VirtualAppliance"),
+					resource.TestCheckResourceAttr(resourceName, "next_hop_in_ip_address", "192.168.0.1"),
+				),
+			},
+			{
+				Config: testAccAzureRMRoute_basic(ri, testLocation()),
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMRouteExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "next_hop_type", "VnetLocal"),
```

Comment:
we should also be able to add in:
```suggestion
					resource.TestCheckResourceAttr(resourceName, "next_hop_type", "VnetLocal"),
					resource.TestCheckResourceAttr(resourceName, "next_hop_in_ip_address", ""),
```

to ensure this field isn't set

Reply:
Good idea. I was being implicit, while being explicit here might be better.